plugin/storage/memory/memory.go

```diff
@@ -111,16 +111,12 @@ func (st *Store) GetDependencies(ctx context.Context, endTs time.Time, lookback
 					if parentSpan.Process.ServiceName == s.Process.ServiceName {
 						continue
 					}
-					depKey := parentSpan.Process.ServiceName + "&&&" + s.Process.ServiceName
-					if _, ok := deps[depKey]; !ok {
-						deps[depKey] = &model.DependencyLink{
-							Parent:    parentSpan.Process.ServiceName,
-							Child:     s.Process.ServiceName,
-							CallCount: 1,
-						}
-					} else {
-						deps[depKey].CallCount++
-					}
+					updateServiceDependencyLinks(parentSpan.Process.ServiceName, s.Process.ServiceName, deps)
+				} else if isClientSpan(s) && !hasCorrespondingServerSpan(s, trace.Spans) {
```

Comment:
why is `else if` appropriate here? A leaf span will have a valid parent so the above branch will be executed, and not this one.